ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala

```diff
@@ -58,16 +58,16 @@ object ParticipantSession {
   def apply(
       partyAllocation: PartyAllocationConfiguration,
       participants: Vector[Channel],
+      maxConnectionAttempts: Int,
       commandInterceptors: Seq[ClientInterceptor],
   )(implicit
       executionContext: ExecutionContext
   ): Future[Vector[ParticipantSession]] =
     Future.traverse(participants) { participant =>
       val services = new LedgerServices(participant, commandInterceptors)
       for {
-        // Keep retrying for about a minute.
         ledgerId <- RetryStrategy
-          .exponentialBackoff(10, 100.millis) { (attempt, wait) =>
+          .exponentialBackoff(attempts = maxConnectionAttempts, 100.millis) { (attempt, wait) =>
```

Comment:
Should we cap this, rather than letting it double every time?

Perhaps to 30 seconds or 1 minute?

Reply:
Yes it does IMO, so that wait times don't become unbearable. I'll try and see if it can be done relatively easily.

Reply:
On second thought, I'd merge as soon as the build passes and tackle it in a subsequent PR.